editor/scene_tree_dock.cpp

```diff
@@ -1503,13 +1503,27 @@ void SceneTreeDock::_script_created(Ref<Script> p_script) {
 		String name = p_script->get_language()->get_global_class_name(p_script->get_path(), NULL, &icon_path);
 		if (ScriptServer::is_global_class(name)) {
 			RES icon = ResourceLoader::load(icon_path);
+			Ref<Script> inherited_script = p_script->get_base_script();
+
+			while (icon.is_null() && inherited_script.is_valid()) {
+				String inherited_icon_path;
+				inherited_script->get_language()->get_global_class_name(inherited_script->get_path(), NULL, &inherited_icon_path);
+				RES inherited_icon = ResourceLoader::load(inherited_icon_path);
+				if (inherited_icon.is_valid())
```

Comment:
Is there a scenario where this may end up never being valid? I'm not at ease with a `while` loop with no guaranteed exit condition.